### DISCUSSION TITLE
Ability to change the OpenAI client that's used on the OpenAI llm

### DISCUSSION THREAD

**Discussion by wricardo** [henomis/lingoose#85] [General]
Feature Suggestion: Ability to Change OpenAI Client on OpenAI llm

I would like to suggest a new feature for the OpenAI llm library that would provide users with the ability to change the OpenAI client used within the library. Currently, the OpenAI client is instantiated during the initialization of the `openAI` struct in the `New` function. However, having the flexibility to use a custom OpenAI client would enhance the versatility and customization options for the library.

Code Snippet:
```go
type openAI struct {
    openAIClient  *openai.Client
    model         Model
....
}

func New(model Model, temperature float32, maxTokens int, verbose bool) *openAI {
    openAIKey := os.Getenv("OPENAI_API_KEY")

    return &openAI{
        openAIClient: openai.NewClient(openAIKey),
...
    }
}

func (o *openAI) WithModel(model Model) *openAI {
    o.model = model
    return o
}

```

Suggested Addition:
To allow users more flexibility and control over the OpenAI client used by the OpenAI llm library, I propose adding a **new** method called `WithClient`. This method would enable users to supply their own instance of the OpenAI client.

Example:

```go
func (o *openAI) WithClient(client *openai.Client) *openAI {
    o.openAIClient = client
    return o
}
```

**Comment by henomis** [comment]
Great! Just inserted as task for the next release [v0.0.8](https://github.com/henomis/lingoose/issues/81)
